src/wallet/rpcwallet.cpp

```diff
@@ -671,11 +671,22 @@ static CAmount GetReceived(const CWallet& wallet, const UniValue& params, bool b
     if (!params[1].isNull())
         min_depth = params[1].get_int();
 
+    bool include_immature{!params[2].isNull() && params[2].get_bool()};
+
+    // Filtering out coinbase outputs is deprecated
+    // It can be enabled by setting deprecatedrpc=filter_coinbase
+    bool filter_coinbase{wallet.chain().rpcEnableDeprecated("filter_coinbase")};
+
     // Tally
     CAmount amount = 0;
     for (const std::pair<const uint256, CWalletTx>& wtx_pair : wallet.mapWallet) {
         const CWalletTx& wtx = wtx_pair.second;
-        if (wtx.IsCoinBase() || !wallet.chain().checkFinalTx(*wtx.tx) || wtx.GetDepthInMainChain() < min_depth) {
+        int depth{wtx.GetDepthInMainChain()};
+        if (depth < min_depth
+            // Coinbase with less than 1 confirmation is an orphan
+            || (wtx.IsCoinBase() && (depth < 1 || filter_coinbase))
+            || (!include_immature && wtx.IsImmatureCoinBase())
```

Comment:
Consistency helps readability. I think you can make this more consistent and readable as:

```suggestion
            || (wtx.IsCoinBase() && (depth < 1 || !include_coinbase))
            || (wtx.IsImmatureCoinBase() && !include_immature_coinbase)
```

(where `include_coinbase = !(wallet.chain().rpcEnableDeprecated("filter_coinbase"))`)

Reply:
Another idea, this logic is close to that added to `ListReceived()` further down in this file; have you considered refactoring this logic into a separate small function? This may make the logic unit-testable, but may not be worth it.

Reply:
@jnewbery Great suggestion, I agree this makes it much more readable.

@LarryRuane Since there are only two instances of this duplication, should we defer to the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming))?